swift/swift.go

```diff
@@ -111,14 +116,19 @@ func (c *Client) GetReader(containerName, objectName string) (io.ReadCloser, htt
 }
 
 // GetReadSeeker creates a read seeker closer to the specified object's data.
-func (c *Client) GetReadSeeker(containerName, objectName string) (io.ReadSeeker, http.Header, error) {
+func (c *Client) GetReadSeeker(containerName, objectName string) (ReadSeekCloser, http.Header, error) {
 	requestData := goosehttp.RequestData{}
 	err := c.touchObject(&requestData, client.HEAD, containerName, objectName)
 	if err != nil {
 		return nil, nil, err
 	}
-	o := &oreadseeker{c, containerName, objectName, requestData.RespLength, 0}
-	return o, requestData.RespHeaders, err
+	return &readSeeker{
```

Comment:
sure. done-ish (named it "object" as swift has objects, not files, and the "swift" seems redundant given that we're in the swift package).